frappe/events_streaming/doctype/event_producer/event_producer.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2019, Frappe Technologies and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Event Producer', {
+	refresh: function(frm) {
+		// formatter for subscribed doctype approval status
+		frm.set_indicator_formatter('status',
+			function(doc) {
+				let indicator = 'orange';
```

Comment:
Alternate way to write this:
```js
return {
	Approved: 'green',
	Rejected: 'red'
}[doc.status] || 'orange';
```

Or you wont have to write this if you make this [method](https://github.com/frappe/frappe/blob/develop/frappe/public/js/frappe/utils/utils.js#L260) smarter by adding "Rejected" text into it.

Reply:
I have added "Rejected" text into it but this `set_indicator_formatter` is for the child table so we will have to retain this method.